jax_triton/pallas/core.py

```diff
@@ -52,19 +56,20 @@ def compute_index(self, *args):
 
 @dataclasses.dataclass(frozen=True)
 class BlockMapping:
-  block_shape: Tuple[int, ...]
+  block_shape: Tuple[Union[Mapped, int], ...]
   index_map_jaxpr: jax_core.ClosedJaxpr
 
   def compute_start_indices(self, loop_idx):
     jaxpr = self.index_map_jaxpr.jaxpr
     consts = self.index_map_jaxpr.consts
     block_indices = jax_core.eval_jaxpr(jaxpr, consts, *loop_idx)
-    return tuple(b * i for b, i in zip(self.block_shape, block_indices))
-
+    return tuple(i if b is mapped else b * i
+                 for b, i in zip(self.block_shape, block_indices))
```

Comment:
Do we have to distinguish between Mapped and a `1` in block_shape?

Reply:
Yea, mapped means *remove* the dimension whereas `1` keeps it around.